AutomatedTesting/Gem/PythonTests/Atom/tests/hydra_AtomEditorComponents_LightAdded.py

```diff
@@ -60,31 +115,61 @@ def AtomEditorComponents_Light_AddedToEntity():
     - Open the "Base" level.
 
     Expected Behavior:
-    The component can be added, used in game mode, hidden/shown, deleted, and has accurate required components.
+    The component can be added, used in game mode, hidden/shown, deleted, all components can be manipulated,
+        and has accurate required components.
     Creation and deletion undo/redo should also work.
 
     Test Steps:
     1) Create a Light entity with no components.
-    2) Add Light component to the Light entity.
-    3) UNDO the entity creation and component addition.
-    4) REDO the entity creation and component addition.
-    5) Cycle through all light types.
-    6) Enter/Exit game mode.
-    7) Test IsHidden.
-    8) Test IsVisible.
-    9) Delete Light entity.
-    10) UNDO deletion.
-    11) REDO deletion.
-    12) Look for errors.
+    2) Remove existing Light components on the entity.
+    3) Add Light component to the Light entity.
+    4) UNDO the entity creation and component addition.
+    5) REDO the entity creation and component addition.
+    6) Set the light type.
+    7) Verify presence of shape component (if applicable).
+    8) Edit the Color parameter.
+    9) Set the Intensity mode parameter.
+    10) Edit the Intensity parameter.
+    11) Set the Attenuation radius Mode.
+    12) Edit the Attenuation radius Radius parameter.
+    13) Enable shadows (if applicable).
+    14) Edit the Shadows Bias parameter.
+    15) Edit the Normal shadow bias parameter.
+    16) Set the Shadowmap size.
+    17) Set the Shadow filter method.
+    18) Edit the Filtering sample count parameter.
+    19) Edit the ESM Exponent parameter.
+    20) Disable Shadows (re-enabled after test for game mode verification).
+    21) Edit the Inner angle parameter.
+    22) Edit the Outer angle parameter.
+    23) Disable Shutters.
+    24) Enable Shutters.
+    25) Enable Both directions.
+    26) Disable Both directions (re-enabled after test for game mode verification).
+    27) Enable Fast approximation.
+    28) Disable Fast approximation.
+    29) Enter/Exit game mode.
+    30) Test IsHidden.
+    31) Test IsVisible.
+    REPEAT tests 2-31 for all applicable light types.
+    32) Delete Light entity.
+    33) UNDO deletion.
+    34) REDO deletion.
+    35) Look for errors.
 
     :return: None
     """
 
     import azlmbr.legacy.general as general
+    import azlmbr.math as math
 
     from editor_python_test_tools.editor_entity_utils import EditorEntity
     from editor_python_test_tools.utils import Report, Tracer, TestHelper
-    from Atom.atom_utils.atom_constants import AtomComponentProperties, LIGHT_TYPES
+    from Atom.atom_utils.atom_constants import (AtomComponentProperties, LIGHT_TYPES, INTENSITY_MODE,
+                                                ATTENUATION_RADIUS_MODE, SHADOWMAP_SIZE, SHADOW_FILTER_METHOD)
+
+    # Set cycle iterations to limit running of tests.
+    iterations = 0
```

Comment:
this seems like needless complication. I would say just cut all this undo redo stuff. it's not needed to test unto adding entities per LIGHT_TYPE.
add the entity
have one for light_type in LIGHT_TYPES loop and do what you need there. I'm not seeing the benefit to having for light_type loops for each separate test section. with branches for entity undo/redo

Reply:
he can just use `LIGHT_TYPES.pop('unknown')` before the test calls and it will be gone from that constant as it iterates

Reply:
Removed and I'm doing the initial component addition/removal/undo outside of the loop.